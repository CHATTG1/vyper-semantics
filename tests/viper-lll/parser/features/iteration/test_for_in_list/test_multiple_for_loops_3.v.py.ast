%pgm(, ,
  %fdecl(%@public, foo, %param(x, %num), %void,
    %assign(%var(p), 0)
    %forlist(i, %list(1 2 3 4),
      %augassign(+=, %var(p), %var(i)))
    %forlist(i, %list(1 2 3 4),
      %augassign(+=, %var(p), %var(i))))
)
