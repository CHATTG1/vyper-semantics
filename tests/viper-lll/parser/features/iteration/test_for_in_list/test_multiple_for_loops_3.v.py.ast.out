$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 3650092561 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $mstore ( 352 , 0 ) , $%seq ( $with ( $_L , 448 , $%seq ( $mstore ( $add ( $mul ( 32 , $uclamplt ( 0 , 4 ) ) , $_L ) , 1 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 1 , 4 ) ) , $_L ) , 2 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 2 , 4 ) ) , $_L ) , 3 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 3 , 4 ) ) , $_L ) , 4 ) ) ) , $repeat ( 416 , 0 , 4 , $%seq ( $mstore ( 384 , $mload ( $add ( 448 , $mul ( $mload ( 416 ) , 32 ) ) ) ) , $%seq ( $with ( $_mloc , 352 , $mstore ( $_mloc , $clamp ( $mload ( 96 ) , $add ( $mload ( $_mloc ) , $mload ( 384 ) ) , $mload ( 64 ) ) ) ) ) ) ) ) , $%seq ( $with ( $_L , 640 , $%seq ( $mstore ( $add ( $mul ( 32 , $uclamplt ( 0 , 4 ) ) , $_L ) , 1 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 1 , 4 ) ) , $_L ) , 2 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 2 , 4 ) ) , $_L ) , 3 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 3 , 4 ) ) , $_L ) , 4 ) ) ) , $repeat ( 608 , 0 , 4 , $%seq ( $mstore ( 576 , $mload ( $add ( 640 , $mul ( $mload ( 608 ) , 32 ) ) ) ) , $%seq ( $with ( $_mloc , 352 , $mstore ( $_mloc , $clamp ( $mload ( 96 ) , $add ( $mload ( $_mloc ) , $mload ( 576 ) ) , $mload ( 64 ) ) ) ) ) ) ) ) , $stop ) ) ) , 0 ) ) )
